src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -180,30 +179,35 @@ def __init__(self,
         else:
             self._reranker = self._DEFAULT_COMPONENTS['reranker']()
 
-        # Normally, index creation params are passed directly to the `.create_canopy_index()` method.  # noqa: E501
-        # However, when KnowledgeBase is initialized from a config file, these params
-        # would be set by the `KnowledgeBase.from_config()` constructor.
-        self._index_params: Dict[str, Any] = {}
+        if pinecone_client:
+            if not isinstance(pinecone_client, Pinecone):
+                raise TypeError(
+                    f"pinecone client must be an instance of Pinecone, "
+                    f"not {type(pinecone_client)}"
+                )
+            self._pinecone_client = pinecone_client
+        else:
+            self._pinecone_client = _get_global_client()
 
         # The index object is initialized lazily, when the user calls `connect()` or
         # `create_canopy_index()`
         self._index: Optional[Index] = None
 
-    def _connect_index(self,
-                       connect_pinecone: bool = True
-                       ) -> None:
-        if connect_pinecone:
-            connect_to_pinecone()
+        self._default_spec = ServerlessSpec(
+            cloud="aws",
+            region="us-west-2"
+        )
```

Comment:
Why do we need to save this as `self.` member variable?  
You can just create it on the fly in `create_canopy_index()` if `spec is None`, can't you?

Reply:
I can actually let me do it

Reply:
Done